A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/SAI/A320_Neo_SAI.js

```diff
@@ -1002,5 +1003,209 @@ class A320_Neo_SAI_PressureIndicator extends HTMLElement {
 }
 customElements.define('a320-neo-sai-pressure-indicator', A320_Neo_SAI_PressureIndicator);
 
+class A320_Neo_SAI_SelfTest extends NavSystemElement {
+    init(root) {
+        this._lastTime = Date.now();
+        this.selfTestElement = this.gps.getChildById("SelfTest");
+        this.complete = false;
+    }
+    onEnter() {
+    }
+    isReady() {
+        return true;
+    }
+    onUpdate(_deltaTime) {
+
+        // Delta time mitigation strategy
+        let curTime = Date.now();
+        let localDeltaTime = curTime - this._lastTime;
+        this._lastTime = curTime;
+
+        const ac_pwr = SimVar.GetSimVarValue("L:ACPowerAvailable", "bool");
+        const dc_pwr = SimVar.GetSimVarValue("L:DCPowerAvailable", "bool");
+        const ColdAndDark = SimVar.GetSimVarValue('L:A32NX_COLD_AND_DARK_SPAWN', 'Bool');
+
+        if ( (ac_pwr || dc_pwr) && !this.complete) {
+            //this.selfTestElement.update(_deltaTime);
+            this.selfTestElement.update(localDeltaTime);
+        }
+        if (!ac_pwr && !dc_pwr) {
+            // TODO: More realistic/advanced Behaviour when power is lost
+            this.selfTestElement.resetTimer();
+        }
+
+        if (!ColdAndDark && ac_pwr && dc_pwr) {
+            // TODO: better way of doing this not on loop
+            this.selfTestElement.finishTest();
+        }
+
+    }
+    onExit() {
+    }
+    onEvent(_event) {
+    }
+}
+class A320_Neo_SAI_SelfTestTimer extends HTMLElement {
+    connectedCallback() {
+        this.construct();
+    }
+
+    construct() {
+
+        this.start_time = 73;
+
+        const boxHeight = 7;
+        const boxWidthSmall = 15;
+        const boxWidth = 18;
+        const boxWidthInit = 30;
+        const boxRow1 = 32.5;
+        const boxRow2 = 46.5;
+        const boxRow3 = 64;
+        const txt_off = 6;
+
+        this.testTimer = Math.floor(Math.random() * 3) + this.start_time;
+        Utils.RemoveAllChildren(this);
+
+        let element = document.querySelector("#SelfTestHider");
```

Comment:
Can be `const`.

```suggestion
        const element = document.querySelector("#SelfTestHider");
```

Also, try finding a more descriptive name ;)

Reply:
Done